### PR TITLE
FOGL-6292 Limit descriptor usage

### PR DESCRIPTION
**Pull request by MarkRiddoch**

### CODE REVIEW

File: C/plugins/storage/sqlite/common/connection_manager.cpp [hunk 1]
```diff
@@ -121,24 +143,42 @@ Connection   *conn;
...
+		if (conn)
+		{
+			inUseLock.lock();
+			inUse.push_front(conn);
+			inUseLock.unlock();
+		}
+		else
+		{
+			std::this_thread::sleep_for(std::chrono::milliseconds(waitfor));
+			if (waitfor < 500)
```
Comment (pintomax) [inline C/plugins/storage/sqlite/common/connection_manager.cpp:178]: Better to add a DEFINE for this

File: C/plugins/storage/sqlite/common/connection_manager.cpp [hunk 2]
```diff
@@ -121,24 +143,42 @@ Connection   *conn;
 Connection *ConnectionManager::allocate()
 {
 Connection *conn = 0;
+int waitfor = 10;
```
Comment (pintomax) [inline C/plugins/storage/sqlite/common/connection_manager.cpp:146]: Better to add a DEFINE

**Review by ashish-jabble — CHANGES_REQUESTED** [review]
With default Sqlite settings. Only 1 south service is running with system info plugin
Now I see overflooded errors in syslogs 
```
Oct 18 16:23:14 aj-ThinkPad-E450 Fledge[14187] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process started: Schedule 'Sys' process 'south_c' task 73091aa4-f045-4bc0-bfb2-0cbbdc190fe6 pid 15622, 1 running tasks#012['services/south_c', '--port=44015', '--address=127.0.0.1', '--name=Sys']
Oct 18 16:23:16 aj-ThinkPad-E450 Fledge[14187] INFO: service_registry: fledge.services.core.service_registry.service_registry: Registered service instance id=78b1e754-8541-4c28-b46c-266a2277994c: <Sys, type=Southbound, protocol=http, address=localhost, service port=38917, management port=45059, status=1>
Oct 18 16:23:28 aj-ThinkPad-E450 Fledge Sys[15624]: WARNING: Current send latency of 9870mS exceeds requested maximum latency of 5000mS
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: attachDb - It was not possible to attach the db :/home/aj/fledge/data/readings_10.db: to the connection :F94B2FF8:, error :too many attached databases - max 10:
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: ReadingsCatalogues error: unknown database readings_10
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: ReadingsCatalogues error: no such table: readings_10.sqlite_master
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: attachDb - It was not possible to attach the db :/home/aj/fledge/data/readings_11.db: to the connection :F94B2FF8:, error :too many attached databases - max 10:
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: ReadingsCatalogues error: unknown database readings_11
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_1: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_1 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_1: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_1 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_1
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_2: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_2 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_2: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_2 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_2
Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_3: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_3 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_3: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_3 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_3
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_4: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_4 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_4: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_4 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_4
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_5: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_5 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_5: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_5 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_5
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_6: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_6 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_6: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_6 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_6
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_7: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_7 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_7: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_7 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_7
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_8: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_8 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_8: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_8 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_8
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_9: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_9 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_9: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_9 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_9
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_10: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_10 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_10: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_10 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_10
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_11: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_11 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 15 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_11: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_11 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_11: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_11 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 23 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_11: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_11 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_11
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_12: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_12 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_12: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_12 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_12
Oct
```

**Comment by MarkRiddoch** [comment]
> With default Sqlite settings. Only 1 south service is running with system info plugin Now I see overflooded errors in syslogs
> 
> ```
> Oct 18 16:23:14 aj-ThinkPad-E450 Fledge[14187] INFO: scheduler: fledge.services.core.scheduler.scheduler: Process started: Schedule 'Sys' process 'south_c' task 73091aa4-f045-4bc0-bfb2-0cbbdc190fe6 pid 15622, 1 running tasks#012['services/south_c', '--port=44015', '--address=127.0.0.1', '--name=Sys']
> Oct 18 16:23:16 aj-ThinkPad-E450 Fledge[14187] INFO: service_registry: fledge.services.core.service_registry.service_registry: Registered service instance id=78b1e754-8541-4c28-b46c-266a2277994c: <Sys, type=Southbound, protocol=http, address=localhost, service port=38917, management port=45059, status=1>
> Oct 18 16:23:28 aj-ThinkPad-E450 Fledge Sys[15624]: WARNING: Current send latency of 9870mS exceeds requested maximum latency of 5000mS
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: attachDb - It was not possible to attach the db :/home/aj/fledge/data/readings_10.db: to the connection :F94B2FF8:, error :too many attached databases - max 10:
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: ReadingsCatalogues error: unknown database readings_10
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: ReadingsCatalogues error: no such table: readings_10.sqlite_master
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: attachDb - It was not possible to attach the db :/home/aj/fledge/data/readings_11.db: to the connection :F94B2FF8:, error :too many attached databases - max 10:
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: ReadingsCatalogues error: unknown database readings_11
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_1: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_1 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_1: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_1 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_1
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_2: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_2 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_2: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_2 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_2
> Oct 18 16:23:44 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_3: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_3 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_3: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_3 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_3
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_4: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_4 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_4: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_4 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_4
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_5: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_5 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_5: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_5 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_5
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_6: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_6 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_6: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_6 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_6
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_7: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_7 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_7: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_7 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_7
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_8: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_8 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_8: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_8 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_8
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_9: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_9 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_9: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_9 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_9
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_10: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_10 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_10: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_10 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_10
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_11: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_11 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 15 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_11: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_11 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_11: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_11 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 23 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_11: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_11 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_11
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_12: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_12 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge message repeated 39 times: [ Storage[14506]: WARNING: SQLPrepare - error :no such table: readings_10.readings_10_12: dbHandle :F94B2FF8: sqlCmd :INSERT INTO  readings_10.readings_10_12 ( id, user_ts, reading ) VALUES  (?,?,?): retry :1: of :40:]
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLPrepare - Database error after maximum retries
> Oct 18 16:23:45 aj-ThinkPad-E450 Fledge Storage[14506]: ERROR: SQLite3 storage plugin raising error: no such table: readings_10.readings_10_12
> Oct
> ```

Hi @ashish-jabble was this a fresh Fledge that you starting or did it already have databases? It looks like it is very confused as to how many reading databases there are?

**Comment by ashish-jabble** [comment]
> Hi @ashish-jabble was this a fresh Fledge that you starting or did it already have databases? It looks like it is very confused as to how many reading databases there are?

On this branch, I did **reset DB**; `make clean; make` & start fledge (i.e so yes fresh DB). 
You can easily reproduce by simply install system info plugin and add south service in enabled state.
And wait for some readings to ingest & then reconfigure asset prefix. You can see those errors.

Below are the DB files on my env after executing above steps
```
$ ls data/*.db*
data/fledge.db      data/readings_1.db-shm  data/readings_2.db-wal  data/readings_4.db      data/readings_5.db-shm  data/readings_6.db-wal
data/fledge.db-shm  data/readings_1.db-wal  data/readings_3.db      data/readings_4.db-shm  data/readings_5.db-wal  data/readings_7.db
data/fledge.db-wal  data/readings_2.db      data/readings_3.db-shm  data/readings_4.db-wal  data/readings_6.db      data/readings_7.db-shm
data/readings_1.db  data/readings_2.db-shm  data/readings_3.db-wal  data/readings_5.db      data/readings_6.db-shm  data/readings_7.db-wal
```

**Pull request closed without merging**